### PR TITLE
Ninja

### PR DESCRIPTION
**Pull request by nattawutk123**

### CODE REVIEW

File: lib/features/home/presentation/home_page.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+import '../../account/presentation/account_page.dart';
+import '../../discover/presentation/discover_page.dart';
+
+class HomePage extends StatefulWidget {
+  const HomePage({super.key});
+
+  @override
+  // ignore: library_private_types_in_public_api
+  _HomePageState createState() => _HomePageState();
```
Comment (GoKubo1997) [inline lib/features/home/presentation/home_page.dart:10]:
```suggestion
  State<HomePage> createState() => _HomePageState();
```

File: lib/features/discover/presentation/discover_page.dart [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import '../../../models/book.dart';
+import '../../../widgets/book_list.dart';
+import '../../../widgets/random_book.dart';
+
+class DiscoverPage extends StatefulWidget {
+  const DiscoverPage({Key? key}) : super(key: key);
+
+  @override
+  // ignore: library_private_types_in_public_api
+  _DiscoverPageState createState() => _DiscoverPageState();
```
Comment (GoKubo1997) [inline lib/features/discover/presentation/discover_page.dart:15]:
```suggestion
  State<DiscoverPage> createState() => _DiscoverPageState();
```

File: lib/models/book.dart [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  Book({
+    required this.id,
+    required this.title,
+    required this.author,
+    required this.category,
+    required this.coverUrl,
+  });
+
+  factory Book.fromJson(Map<String, dynamic> json) {
+    return Book(
```
Comment (GoKubo1997) [inline lib/models/book.dart:17]: Cool.

File: lib/widgets/book_list.dart [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                          SizedBox(
+                            height: 30, //[MODIFY]
+                            width: 97,
+                            child: Column(
+                              crossAxisAlignment: CrossAxisAlignment.start,
+                              children: [
+                                Text(
+                                  book.title.length > 10
+                                      ? '${book.title.substring(0, 9)}...'
+                                      : book.title,
```
Comment (GoKubo1997) [inline lib/widgets/book_list.dart:68]: You also can set the maxiLine for the Text widget. If the text gets longer for a given area, the "..." will be shown.

```dart
Text(
  'A long sentence',
  maxLines: 1,
)
```

File: lib/widgets/book_list.dart [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+                                  style: const TextStyle(
+                                    fontSize: 12,
+                                    fontWeight: FontWeight.bold,
+                                  ),
+                                ),
+                                const SizedBox(height: 2),
+                                Text(
+                                  book.author.length > 10
+                                      ? '${book.author.substring(0, 9)}...'
+                                      : book.author,
```
Comment (GoKubo1997) [inline lib/widgets/book_list.dart:78]: Here as well.

File: lib/widgets/random_book.dart [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+  @override
+  void initState() {
+    super.initState();
+    loadBooks();
+  }
+
+  void loadBooks() async {
+    final jsonFile = await rootBundle.loadString('assets/books.json');
+    final parsedJson = json.decode(jsonFile);
```
Comment (GoKubo1997) [inline lib/widgets/random_book.dart:26]: This operation is done in the _DiscoverPageState class as well. So you want to use the existing `List<Book> _books` in the  _DiscoverPageState class. To do it, you need to take the list with the constructor of the RandomBookWidget class.

File: lib/features/discover/presentation/discover_page.dart [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+  @override
+  void initState() {
+    super.initState();
+    _loadBooks();
+  }
+
+  void _loadBooks() async {
+    final jsonFile = await rootBundle.loadString('assets/books.json');
+    final parsedJson = json.decode(jsonFile);
```
Comment (GoKubo1997) [inline lib/features/discover/presentation/discover_page.dart:29]: Cool.

**Review by GoKubo1997 — COMMENTED** [review]
Looks solid overall.

Please make sure to make changes on the review02 branch.

File: lib/widgets/book_list.dart (hunk 4 above)
Comment (nattawutk123) [reply lib/widgets/book_list.dart:68]: Thank you, That really helpful.

**Pull request merged**